src/theory/arith/nonlinear_extension.cpp

```diff
@@ -3360,6 +3360,53 @@ std::vector<Node> NonlinearExtension::checkTangentPlanes() {
                       << "Tangent plane lemma : " << tlem << std::endl;
                   lemmas.push_back(tlem);
                 }
+
+                // tangent plane reverse implication
+
+                // t <= tplane -> ( (a <= a_v ^ b >= b_v) v (a >= a_v ^ b <=
+                // b_v) ) in clause form, the above becomes t <= tplane -> a <=
+                // a_v v b <= b_v t <= tplane -> b >= b_v v a >= a_v
+                Node a_leq_av = NodeManager::currentNM()->mkNode(LEQ, a, a_v);
```

Comment:
Can you cache `NodeManager * nm = NodeManager::currentNM();` at the top of this function and use `nm->mkNode` throughout? (This is the new style we've been conforming to; my older code above did not conform to this style, but don't worry about changing it).